Orange/widgets/visualize/tests/test_owscatterplot.py

```diff
@@ -392,6 +392,27 @@ def test_vizrank(self):
         states = [state for state in vizrank.iterate_states(None)]
         self.assertIsNotNone(vizrank.compute_score(states[0]))
 
+    def test_vizrank_class_nan(self):
+        """
+        When class values are nan, vizrank should be disabled. It should behave like
+        the class column is missing.
+        GH-2757
+        """
+        def assert_vizrank_enabled(data, is_enabled):
+            self.send_signal(self.widget.Inputs.data, data)
+            self.assertEqual(is_enabled, self.widget.vizrank_button.isEnabled())
+
+        data1 = Table("iris")[::30]
+        data2 = Table("iris")[::30]
+        data2.Y[:] = np.nan
+        domain = Domain(
+            attributes=data2.domain.attributes[:4], class_vars=DiscreteVariable("iris", values=[]))
+        data2 = Table(domain, data2.X, Y=data2.Y)
+        data3 = Table("iris")[::30]
```

Comment:
What is the difference between data1 and data3?  (i.e. why this line)

Reply:
Now it is the difference.